File: speech_to_text_app/lib/data/models/api/responses/todo/todo.dart
class Todo {
  final String id;
  final String transcriptionId;
  final String title;
  final String? description;
  String? audioTitle;
  bool? isCompleted;

  Todo({
    required this.id,
    required this.transcriptionId,
    required this.title,
    this.audioTitle,
    this.description,
    this.isCompleted,
  });

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json['id'],
      transcriptionId: json['transcription_id'],
      title: json['title'],
      audioTitle: json['audio_title'] ?? '',
      description: json['description'],
      isCompleted: json['is_completed'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'transcription_id': transcriptionId,
      'title': title,
      'audio_title': audioTitle,
      'description': description,
      'is_completed': isCompleted,
    };
  }

  Todo copyWith({
    String? id,
    String? transcriptionId,
    String? title,
    String? audioTitle,
    String? description,
    bool? isCompleted,
  }) {
    return Todo(
      id: id ?? this.id,
      transcriptionId: transcriptionId ?? this.transcriptionId,
      title: title ?? this.title,
      audioTitle: audioTitle ?? this.audioTitle,
      description: description ?? this.description,
      isCompleted: isCompleted ?? this.isCompleted,
    );
  }
}

File: server/pydantic_schemas/todo.py
# pydantic_schemas/todo.py
from pydantic import BaseModel
from typing import Optional

class TodoBase(BaseModel):
    title: str
    description: Optional[str] = None

class TodoCreate(TodoBase):
    pass

class TodoResponse(TodoBase):
    id: str
    transcription_id: str


class TodoWithAudioResponse(BaseModel):
    id: str
    transcription_id: str
    title: str
    description: str
    is_completed: bool
    audio_title: str

class TodoUpdate(TodoBase):
    title: str = None
    description: str = None
    is_completed: bool  = None

class TodoCreate2(BaseModel):
    transcription_id: str
    title: str
    description: str
    is_completed: bool = False
    class Config:
        orm_mode = True
